pkg/ccl/backupccl/backup_test.go

```diff
@@ -7326,6 +7326,55 @@ func TestCleanupDoesNotDeleteParentsWithChildObjects(t *testing.T) {
 	})
 }
 
+func TestManifestTooNew(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	_, _, sqlDB, rawDir, cleanupFn := BackupRestoreTestSetup(t, singleNode, 1, InitManualReplication)
+	defer cleanupFn()
+	sqlDB.Exec(t, `CREATE DATABASE r1`)
+	sqlDB.Exec(t, `BACKUP DATABASE r1 TO 'nodelocal://0/too_new'`)
+	sqlDB.Exec(t, `DROP DATABASE r1`)
+	// Prove we can restore.
+	sqlDB.Exec(t, `RESTORE DATABASE r1 FROM 'nodelocal://0/too_new'`)
+	sqlDB.Exec(t, `DROP DATABASE r1`)
+
+	// Load/deserialize the manifest so we can mess with it.
+	manifestPath := filepath.Join(rawDir, "too_new", backupManifestName)
+	manifestData, err := ioutil.ReadFile(manifestPath)
+	require.NoError(t, err)
+	manifestData, err = decompressData(manifestData)
+	require.NoError(t, err)
+	var backupManifest BackupManifest
+	require.NoError(t, protoutil.Unmarshal(manifestData, &backupManifest))
+
+	// Bump the version and write it back out to make it look newer.
+	backupManifest.ClusterVersion = roachpb.Version{Major: 99, Minor: 1}
+	manifestData, err = protoutil.Marshal(&backupManifest)
+	require.NoError(t, err)
+	require.NoError(t, ioutil.WriteFile(manifestPath, manifestData, 0644 /* perm */))
+	// Also write the checksum file to match the new manifest.
+	checksum, err := getChecksum(manifestData)
+	require.NoError(t, err)
+	require.NoError(t, ioutil.WriteFile(manifestPath+backupManifestChecksumSuffix, checksum, 0644 /* perm */))
+
+	// Verify we reject it.
+	sqlDB.ExpectErr(t, "backup from version 99.1 is newer than current version", `RESTORE DATABASE r1 FROM 'nodelocal://0/too_new'`)
+
+	// Bump the version down and write it back out to make it look older.
+	backupManifest.ClusterVersion = roachpb.Version{Major: 20, Minor: 2, Internal: 2}
```

Comment:
Can we also add a test case where we nil out the cluster version to simulate earlier backups?

Reply:
good idea. Done.